sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -95,18 +99,59 @@ public ReactorConnection(String connectionId, ConnectionOptions connectionOption
         this.handler = handlerProvider.createConnectionHandler(connectionId, connectionOptions);
 
         this.retryPolicy = RetryUtil.getRetryPolicy(connectionOptions.getRetry());
+        this.operationTimeout = connectionOptions.getRetry().getTryTimeout();
         this.senderSettleMode = senderSettleMode;
         this.receiverSettleMode = receiverSettleMode;
 
-        this.connectionMono = Mono.fromCallable(this::getOrCreateConnection);
+        this.connectionMono = Mono.fromCallable(this::getOrCreateConnection)
+            .flatMap(reactorConnection -> {
+                final Mono<AmqpEndpointState> activeEndpoint = getEndpointStates()
+                    .filter(state -> state == AmqpEndpointState.ACTIVE)
+                    .next()
+                    .timeout(operationTimeout, Mono.error(new AmqpException(false, String.format(
+                        "Connection '%s' not opened within AmqpRetryOptions.tryTimeout(): %s", connectionId,
+                        operationTimeout), handler.getErrorContext())));
+                return activeEndpoint.thenReturn(reactorConnection);
+            })
+            .or(onClosedError("Could not get active connection."))
+            .doOnError(error -> {
+                final String message = String.format(
+                    "connectionId[%s] Error occurred while connection was starting. Error: %s", connectionId, error);
+
+                if (isDisposed.getAndSet(true)) {
+                    logger.verbose("connectionId[{}] was already disposed. {}", connectionId, message);
+                } else {
+                    dispose(new AmqpShutdownSignal(false, false, message));
```

Comment:
The ConnectionHandler is bound to a single connection, so it will always be the same one. When we create a proton-j Connection, we bind an instance of the ConnectionHandler to it. So each connection has its own connection handler instance.